extensions/wikia/Track/Track.php

```diff
@@ -1,59 +1,130 @@
+			   ( $wgTitle ? $sep . 'n=' . $wgTitle->getNamespace() : '' ) .
+			   ( !empty( $wgAdServerTest ) ? $sep . 'db_test=1' : '' );
 
 		// Handle any parameters passed to us
-		if ($param) {
-			foreach ($param as $key => $val) {
-				$url .= $sep.urlencode($key).'='.urlencode($val);
+		if ( $param ) {
+			foreach ( $param as $key => $val ) {
+				$url .= $sep . urlencode( $key ) . '=' . urlencode( $val );
 			}
 		}
 
 		return $url;
 	}
 
-	private static function getViewJS ($param=null) {
+	private static function getGAURL( $type, $category, $action, $label, $tid, $extraParams = [ ], $for_html = null ) {
+		global $wgTitle, $wgContLanguageCode, $wgDBname, $wgUser, $wgCityId, $wgGAUserIdSalt;
+
+		$skinName = RequestContext::getMain()->getSkin()->getSkinName();
+		$adContext = ( new AdEngine2ContextService() )->getContext( $wgTitle, $skinName );
+		$hubFactory = WikiFactoryHub::getInstance();
+
+		$params = [
+			'v' => static::GA_VERSION,
+			'ds' => 'backend',
+			't' => $type,
+			'ni' => 1,
+			'dl' => $wgTitle->getFullURL(),
+			'ul' => $wgContLanguageCode,
+			'de' => 'UTF-8',
+			'dt' => $wgTitle->getText(),
+			'ec' => $category,
+			'ea' => $action,
+			'el' => $label,
+			'_utma' => $_COOKIE[ '__utma' ] ?? '',
+			'_utmz' => $_COOKIE[ '__utmz' ] ?? '',
+			'cid' => $_COOKIE[ '_ga' ] ? explode( ".", $_COOKIE[ '_ga' ] )[ 2 ] : uniqid(),
+			'tid' => $tid,
+			// custom dimensions
+			'cd1' => $wgDBname,
+			'cd2' => $wgContLanguageCode,
+			'cd3' => implode( ',', $hubFactory->getWikiVertical( $wgCityId ) ),
+			'cd4' => $skinName,
+			'cd5' => $wgUser->isAnon() ? 'anon' : 'user',
+			'cd8' => WikiaPageType::getPageType(),
+			'cd9' => $wgCityId,
+			'cd13' => AdTargeting::getEsrbRating(),
+			'cd14' => isset( $adContext[ 'opts' ][ 'showAds' ] ) ? 'Yes' : 'No',
+			'cd15' => WikiaPageType::isCorporatePage(),
+			'cd17' => implode( ',', $hubFactory->getWikiVertical( $wgCityId ) ),
+			'cd18' => implode( ',', $hubFactory->getWikiCategories( $wgCityId ) ),
+			'cd19' => WikiaPageType::getArticleType(),
+			'cd21' => $wgTitle->getArticleID(),
+			'cd23' => $wgUser->isSpecificPowerUser( Wikia\PowerUser\PowerUser::TYPE_FREQUENT ) ? 'Yes' : 'No',
+			'cd24' => $wgUser->isSpecificPowerUser( Wikia\PowerUser\PowerUser::TYPE_LIFETIME ) ? 'Yes' : 'No',
+			'cd25' => $wgTitle->getNamespace(),
+		];
+		if ( !$wgUser->isAnon() ) {
+			$params[ 'uid' ] = md5( $wgUser->getId() . $wgGAUserIdSalt );
+		}
+		$params = array_merge( $params, $extraParams );
+		$url = static::GA_URL . '/debug/collect?' .
+			   implode( $for_html ? '&amp;' : '&', array_map( function ( $k, $v ) {
+				   return "{$k}={$v}";
+			   }, array_keys( $params ), $params ) );
+
+		return $url;
+	}
+
+	private static function getGATrackingIds() {
+		global $wgDevelEnvironment, $wgIsGASpecialWiki, $wgUser;
+
+		$tids = [ ];
+		$tids[] = $wgDevelEnvironment ? 'UA-32129070-2' : 'UA-32129070-1';
```

Comment:
In this case we either need to rethink the approach of tracking on both front- and back-end, or introduce sampling here - otherwise we'll make GA reports for anons useless